packages/datadog-core/src/storage/async_resource.js

```diff
@@ -49,7 +53,13 @@ class AsyncResourceStorage {
 
   _createHook () {
     return createHook({
-      init: this._init.bind(this)
+      init: this._init.bind(this),
+      before () {
```

Comment:
For this to work properly in older versions, it needs to also work with `AsyncHooksStorage`. This should be easy by returning the object from `_createHook` instead of the hook, and then calling `createHook` on the result of `_createHook` so that `AsyncHooksStorage` can add its destroy hook to the object.